plugins/anomaly-detection/AnomalyDetection.py

```diff
@@ -0,0 +1,615 @@
+                            "flowStartSeconds"])
+    plotDF5 = plotDF4.join(plotDF2,
+                           ["sourceIP", "sourceTransportPort", "destinationIP",
+                            "destinationTransportPort", "protocolIdentifier",
+                            "flowStartSeconds"])
+    if algo_type == "EWMA":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_ewma,
+                                calculate_ewma_anomaly, tad_id_input)
+    elif algo_type == "ARIMA":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_arima,
+                                calculate_arima_anomaly, tad_id_input)
+    elif algo_type == "DBSCAN":
+        ret_plot = plot_anomaly(spark, plotDF5, algo_type, calculate_dbscan,
+                                calculate_dbscan_anomaly, tad_id_input)
+    return spark, ret_plot
+
+
+def write_anomaly_detection_result(
+        result_df, db_jdbc_address, result_table_name, tad_id_input):
+    if not tad_id_input:
+        tad_id = str(uuid.uuid4())
+    else:
+        tad_id = tad_id_input
+
+    result_df.write.mode("append").format("jdbc").option(
+        "driver", "ru.yandex.clickhouse.ClickHouseDriver").option(
+        "url", db_jdbc_address).option(
+        "user", os.getenv("CH_USERNAME")).option(
+        "password", os.getenv("CH_PASSWORD")).option(
+        "dbtable", result_table_name).save()
+    return tad_id
+
+
+def main():
+    db_jdbc_address = (
+        "jdbc:clickhouse://clickhouse-clickhouse.flow-visibility.svc:8123")
+    result_table_name = "default.tadetector"
+    algo_type = ""
+    start_time = ""
+    end_time = ""
+    tad_id_input = None
+    help_message = """
+    Start the policy recommendation spark job.
+        Options:
+        -h, --help: Show help message.
+        -t, --type=EWMA: Type argument describes the anomaly detection Algo 
+            to use.Currently Supported Algos are EWMA, ARIMA and DBSCAN
+        -d, --db_jdbc_url=None: The JDBC URL used by Spark jobs connect to 
+            the ClickHouse database for reading flow records and writing 
+            result. jdbc:clickhouse://clickhouse-clickhouse.flow-visibility
+            .svc:8123 is the ClickHouse JDBC URL used by default.
+        -s, --start_time=None: The start time of the flow records 
+            considered for the throughput Anomaly Detection. Format is 
+            YYYY-MM-DD hh:mm:ss in UTC timezone. Default value is None,
+            which means no limit of the start time of flow records.
+        -e, --end_time=None: The end time of the flow records considered 
+            for thethroughput Anomaly Detection.
+            Format is YYYY-MM-DD hh:mm:ss in UTC timezone.
+            Default value is None,which means no limit of the end time 
+            of flow records.
+        -i, --id=None: Recommendation job ID in UUID format. If not 
+            specified, it will be generated automatically.
+
+        Usage Example:
```

Comment:
I feel this `Usage Example` could be removed because we don't expect user trying this .py file directly

Reply:
ack